phone-book/src/app.js

```diff
@@ -0,0 +1,177 @@
+class App {
+  constructor(appContainerId) {
+    this.state = {};
+    this.appContainer = document.getElementById(appContainerId);
+    this.router = new Router(this);
+    this.serverAPI = new ServerAPI();
+    const userPage = new EditUser(this);
+    this.pages = {
+      contacts : {
+        pageObject : new Contacts(this),
+        href : '/contacts.html'
+      },
+      keypad : {
+        pageObject : new Keypad(this),
+        href : '/keypad.html'
+      },
+      editUser : {
+        pageObject : userPage,
+        href : '/edit-contact.html'
+      },
+      user : {
+        pageObject : new User(this),
+        href : '/user.html'
+      },
+      addUser : {
+        pageObject : userPage,
+        href : '/add-user.html'
+      }
+    }
+
+    const appHtml = this.renderAppContainer();
+    this.appContainer.innerHTML = appHtml;
+    this.addRouter();
+
+    this.state.pageName = 'Contacts';
+    this.updateState({activePage : 'contacts'});
+    this.router.gotToPage(this.state,'/index.html');
```

Comment:
probably we don't need to tell the router what page to render, it could just render an active page, or "default-page"
'index.html' - is just a default page in your case.

Furthermore App should don't know anything about router methods. It could just initialize router and Router should care about everything else